webpack/assets/javascripts/react_app/components/common/ResourceSwitcher/index.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
```

Comment:
just wondering, as i didn't fully catched up, is this structure in line with @sharvit proposal ?

Reply:
no... it's not a redux connected component.